### DISCUSSION TITLE
Make Unnecessary Clicks Count as Misses (No Spam)

### DISCUSSION THREAD

**Discussion by Oscujic** [ppy/osu#30770] [Ideas]
**Description:**
- When the player clicks but is not on top of a clickable object, this should count as a miss, breaking combo and playing a "miss" sound (Similar to the the "Jigsaw 1" mod in [McOsu](https://github.com/McKay42/McOsu)).

**Reason:**
- The general effect will be to punish spamming (the most common form of which is misjudging the number of circles in a stream or stack), giving greater feedback and raising the skill ceiling.

**Implementation Details:**
- Unnecessary clicks can be accounted for in scoring by adding them to the denominator of the accuracy calculation, as the "false positives" of a [Critical Success Index](https://resources.eumetrain.org/data/4/451/english/msg/ver_categ_forec/uos2/uos2_ko4.htm).
- Unnecessary clicks should be shown separately from regular misses on the final score card.
- To prevent punishing "practice taps", the check for unnecessary clicks should only be done when the time until or since the chronologically closest object is within the map's approach time.

**Alternatives:**
- If it is determined that it is not suitable for this feature to be made standard for the game, it could be made an optional mod ("No Spam"), preferably with no score modifier, so those who want the additional feedback can have it, while still being able to set scores as normal.

**Comment by mixalisnew** [comment]
That sounds annoying and pointless in my opinion. I dont see why i should get punished for over tapping a stream by 1 note or something

**Reply by Oscujic** [reply]
That's the point, though. It makes it all the more rewarding when you tap just right.

**Reply by rvnexe** [reply]
this can be implemented as a mod, but it will be horrible if used as a core gameplay rule.

**Comment by hhirtz** [comment]
Coming from https://github.com/ppy/osu/discussions/34309 (same idea but for mania)

I agree this better not be a core gameplay mechanic, but a mod that does this, for mania at least, would replicate the behavior of other rhythm games (pop'n music for example) where you cannot hit a key without a note being there. It would allow mania players to better compare with these other games and also increase the skill level for those that want it as oscujic said.

I'm also curious as to where in maps with tricky timings I'm hitting keys where I shouldn't.

Anyways there's a lot of arguments in favor for a mod that does this.

**Comment by cerfeque** [comment]
This is the most disguting thing i ever heard. It sounds good only for the first 5 seconds before you think about it for a little more and realize every player doubletaps, especially with RT keyboard. Upload ANY of your replays to o!rdr, look at the red lines on UR bar. Yeah, that's doubletaps. You probably never noticed it. Doesn't look so fun anymore?
